network/endpoint_linux.go

```diff
@@ -53,205 +54,200 @@ func (nw *network) newEndpointImpl(
-			vlanid = epInfo.Data[VlanIDKey].(int)
+	if defaultEpInfo.Data != nil {
+		if _, ok := defaultEpInfo.Data[VlanIDKey]; ok {
+			vlanid = defaultEpInfo.Data[VlanIDKey].(int)
 		}
 
-		if _, ok := epInfo.Data[LocalIPKey]; ok {
-			localIP = epInfo.Data[LocalIPKey].(string)
+		if _, ok := defaultEpInfo.Data[LocalIPKey]; ok {
+			localIP = defaultEpInfo.Data[LocalIPKey].(string)
 		}
 	}
 
-	if _, ok := epInfo.Data[OptVethName]; ok {
-		key := epInfo.Data[OptVethName].(string)
+	if _, ok := defaultEpInfo.Data[OptVethName]; ok {
+		key := defaultEpInfo.Data[OptVethName].(string)
 		logger.Info("Generate veth name based on the key provided", zap.String("key", key))
 		vethname := generateVethName(key)
 		hostIfName = fmt.Sprintf("%s%s", hostVEthInterfacePrefix, vethname)
 		contIfName = fmt.Sprintf("%s%s2", hostVEthInterfacePrefix, vethname)
 	} else {
 		// Create a veth pair.
 		logger.Info("Generate veth name based on endpoint id")
-		hostIfName = fmt.Sprintf("%s%s", hostVEthInterfacePrefix, epInfo.Id[:7])
-		contIfName = fmt.Sprintf("%s%s-2", hostVEthInterfacePrefix, epInfo.Id[:7])
+		hostIfName = fmt.Sprintf("%s%s", hostVEthInterfacePrefix, defaultEpInfo.Id[:7])
+		contIfName = fmt.Sprintf("%s%s-2", hostVEthInterfacePrefix, defaultEpInfo.Id[:7])
 	}
 
-	// epClient is non-nil only when the endpoint is created for the unit test.
-	if epClient == nil {
-		//nolint:gocritic
-		if vlanid != 0 {
-			if nw.Mode == opModeTransparentVlan {
-				logger.Info("Transparent vlan client")
-				if _, ok := epInfo.Data[SnatBridgeIPKey]; ok {
-					nw.SnatBridgeIP = epInfo.Data[SnatBridgeIPKey].(string)
+	ep := &endpoint{
+		Id:                       defaultEpInfo.Id,
+		IfName:                   contIfName, // container veth pair name. In cnm, we won't rename this and docker expects veth name.
+		HostIfName:               hostIfName,
+		InfraVnetIP:              defaultEpInfo.InfraVnetIP,
+		LocalIP:                  localIP,
+		IPAddresses:              defaultEpInfo.IPAddresses,
+		DNS:                      defaultEpInfo.DNS,
+		VlanID:                   vlanid,
+		EnableSnatOnHost:         defaultEpInfo.EnableSnatOnHost,
+		EnableInfraVnet:          defaultEpInfo.EnableInfraVnet,
+		EnableMultitenancy:       defaultEpInfo.EnableMultiTenancy,
+		AllowInboundFromHostToNC: defaultEpInfo.AllowInboundFromHostToNC,
+		AllowInboundFromNCToHost: defaultEpInfo.AllowInboundFromNCToHost,
+		NetworkNameSpace:         defaultEpInfo.NetNsPath,
+		ContainerID:              defaultEpInfo.ContainerID,
+		PODName:                  defaultEpInfo.PODName,
+		PODNameSpace:             defaultEpInfo.PODNameSpace,
+		Routes:                   defaultEpInfo.Routes,
+		SecondaryInterfaces:      make(map[string]*InterfaceInfo),
+	}
+	if nw.extIf != nil {
+		ep.Gateways = []net.IP{nw.extIf.IPv4Gateway}
+	}
+
+	for _, epInfo := range epInfo {
+		// testEpClient is non-nil only when the endpoint is created for the unit test
+		epClient := testEpClient
```

Comment:
why does it require another variable?

Reply:
we want to use the testEpClient if it's not nil, otherwise we'll set it to nil and create new endpoint client when we loop through epInfo

Reply:
I dont get it.. no need to rename epClient to testCpClient then

Reply:
added comment